src/System.Net.Sockets/tests/FunctionalTests/SelectTest.cs

```diff
@@ -238,5 +238,69 @@ private static void DisposeSockets(IEnumerable<KeyValuePair<Socket, Socket>> soc
                 pair.Value.Dispose();
             }
         }
+
+        [OuterLoop]
+        [Fact]
+        public static void Select_AcceptNonBlocking_Success()
+        {
+            using (Socket listenSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                int port = listenSocket.BindToAnonymousPort(IPAddress.Loopback);
+
+                listenSocket.Blocking = false;
+
+                listenSocket.Listen(5);
+
+                Task t = Task.Run(() => { DoAccept(listenSocket, 5); });
+
+                // Loop, doing connections and pausing between
+                for (int i = 0; i < 5; i++)
+                {
+                    Thread.Sleep(50);
+                    using (Socket connectSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+                    {
+                        connectSocket.Connect(listenSocket.LocalEndPoint);
+                    }
+                }
+
+                // Give the task 5 seconds to complete; if not, assume it's hung.
+                bool completed = t.Wait(5000);
```

Comment:
We may find we need to increase this timeout, as I think we've hit such timeouts under load in CI.